core/chain_util.go

```diff
@@ -52,8 +53,20 @@ func CalcDifficulty(time, parentTime uint64, parentDiff *big.Int) *big.Int {
 		diff.Sub(parentDiff, adjust)
 	}
 	if diff.Cmp(params.MinimumDifficulty) < 0 {
-		return params.MinimumDifficulty
+		diff = params.MinimumDifficulty
 	}
+
+	periodCount := new(big.Int).Add(number, big.NewInt(1))
+	periodCount.Div(periodCount, expDiffPeriod)
+	if periodCount.Cmp(big.NewInt(1)) > 0 {
+		// diff = diff - 2^(periodCount - 1)
```

Comment:
The exponential factor should be added, not subtracted, right? This makes the diff exponentially smaller. after expDiffPeriod